### DISCUSSION TITLE
Rocket development improvements

### DISCUSSION THREAD

**Discussion by rdiaz82** [boostercloud/booster#532] [Meta]
After the development of the [AWS MQTT rocket ](https://github.com/rdiaz82/rocket-mqtt-aws-infrastructure)I have detected some details that can be polished to make rockets development smoother: 

**1. Include the booster config into the `mount` method:** Normally you need to have access to Booster project properties and right now we need to pass manually the Booster configuration into the rocket initialization. 

**2. Creation of a rocket utils library:** If we want to do anything related to a Booster application like inserting an event we need to add Booster packages as dependencies. The problem with this is that the size of the rocket package is impacted because normally the Booster packages require the AWS CDK which is huge. It would be nice to have a slim package that solving commons booster project interactions in a way of utils library that can be called from the rocket. 

**3. Utilities to interact with the user project**: Right now if the rocket needs to create an asset into the user project like an entity, read model, etc... the user has to create them manually. It would be nice if we create a mechanism to allow us to interact with the user project from the rocket to create assets inside the user project without the user interaction. 

**4.  Improve the error logging**. Right now if there is any problem during the CDK interaction between the rocket and the booster project, the CLI simply fails and doesn't show any clue about what has failed.

**Comment by charlietfe** [comment]
**5. Improve the output**. Directly related to that we should standardize the generated outputs as a result of the deployed infrastructure. Now we rely heavily on AWS Cloud Formation outputs (at least in the AWS provider) so I think we should create an output file as a result of the deployment in a readable format. In that way the user, even if is an old deployment, might query that file to verify the generated APIs.
